test/unit/org/apache/cassandra/index/sai/cql/LiteralTermsDistributionTest.java

```diff
@@ -0,0 +1,139 @@
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
+import org.apache.cassandra.index.sai.memory.TrieMemtableIndex;
+import org.apache.cassandra.index.sai.plan.Expression;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * This test aims to cover the different encodings of terms distribution to ensure expected results.
+ */
+@RunWith(Parameterized.class)
+public class LiteralTermsDistributionTest extends SAITester
+{
+
+    @Parameterized.Parameter
+    public CQL3Type.Native testType;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data()
+    {
+        return List.of(new Object[]{ CQL3Type.Native.ASCII }, new Object[] { CQL3Type.Native.TEXT });
+    }
+
+    private StorageAttachedIndex getIndex()
+    {
+        // A bit brittle, but this is the most realistic way to test encodings.
+        return (StorageAttachedIndex) Keyspace.open(KEYSPACE).getColumnFamilyStore(currentTable()).getIndexManager()
+                                              .getIndexByName(currentIndex());
+    }
+
+    @Test
+    public void testLiteralIndexEstimates() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int primary key, a " + testType + ')');
+        createIndex("CREATE CUSTOM INDEX ON %s(a) USING 'StorageAttachedIndex'");
+        waitForIndexQueryable();
+
+        execute("INSERT INTO %s (pk, a) VALUES (1, 'a')");
+        execute("INSERT INTO %s (pk, a) VALUES (2, 'aa')");
+        execute("INSERT INTO %s (pk, a) VALUES (3, 'ab')");
+        execute("INSERT INTO %s (pk, a) VALUES (4, 'c')");
+        execute("INSERT INTO %s (pk, a) VALUES (5, 'z')");
+        execute("INSERT INTO %s (pk, a) VALUES (6, 'zz')");
+
+        var sai = getIndex();
+
+        assertInMemoryEstimateCount(sai, Operator.EQ, "ab", 1);
+        assertInMemoryEstimateCount(sai, Operator.LT, "ab", 2, 1);
+        assertInMemoryEstimateCount(sai, Operator.GT, "ab", 3, 1);
+        assertInMemoryEstimateCount(sai, Operator.LTE, "ab", 3);
+        assertInMemoryEstimateCount(sai, Operator.GTE, "ab", 4);
+        assertInMemoryEstimateCount(sai, Operator.EQ, "x", 0);
+
+        flush();
+
+        assertSSTableEstimateCount(sai, Operator.EQ, "ab", 1);
+        assertSSTableEstimateCount(sai, Operator.LT, "ab", 2, 1);
+        assertSSTableEstimateCount(sai, Operator.GT, "ab", 3, 1);
+        assertSSTableEstimateCount(sai, Operator.LTE, "ab", 3);
+        assertSSTableEstimateCount(sai, Operator.GTE, "ab", 4);
+        assertSSTableEstimateCount(sai, Operator.EQ, "x", 0);
+    }
+
+
+    private void assertInMemoryEstimateCount(StorageAttachedIndex index, Operator op, String value, long expectedCount)
+    {
+        assertInMemoryEstimateCount(index, op, value, expectedCount, 0);
+    }
+
+    private void assertInMemoryEstimateCount(StorageAttachedIndex index, Operator op, String value, long expectedCount, long roundingValue)
```

Comment:
Nit: you don't need a rounding value here. That was only necessary because of the logic from truncated decimal and varint types.